zerver/views/presence.py

```diff
@@ -68,13 +70,36 @@ def update_user_status_backend(
     user_profile: UserProfile,
     away: Optional[bool] = REQ(json_validator=check_bool, default=None),
     status_text: Optional[str] = REQ(str_validator=check_capped_string(60), default=None),
+    emoji_name: Optional[str] = REQ(default=None),
+    emoji_code: Optional[str] = REQ(default=None),
+    reaction_type: Optional[str] = REQ(default=None),
 ) -> HttpResponse:
 
     if status_text is not None:
         status_text = status_text.strip()
 
-    if (away is None) and (status_text is None):
+    if (away is None) and (status_text is None) and (emoji_name is None):
         raise JsonableError(_("Client did not pass any new values."))
+    if emoji_name == "":
+        # Reset the emoji_code and reaction_type if emoji_name is empty.
+        emoji_code = ""
+        reaction_type = UserStatus.UNICODE_EMOJI
+
+    elif emoji_name is not None:
+        if emoji_code is None:
+            # The emoji_code argument is only required for rare corner
+            # cases discussed in the long block comment below.  For simple
+            # API clients, we allow specifying just the name, and just
+            # look up the code using the current name->code mapping.
+            emoji_code = emoji_name_to_emoji_code(user_profile.realm, emoji_name)[0]
+
+        if reaction_type is None:
+            reaction_type = emoji_name_to_emoji_code(user_profile.realm, emoji_name)[1]
+
+    elif reaction_type or emoji_code:
+        raise JsonableError(
+            _("Client must pass emoji_name if they pass either emoji_code or reaction_type.")
+        )
```

Comment:
This should be calling `check_emoji_request` to validate that the emoji requested is valid.  I can add that while integrating this.